### PR TITLE
Feat xxx extract epic

### PR DESCRIPTION
**Pull request by lvalerom**
This a new feature in the XMM Newton module that allows to extract the EPIC images for different bands and instruments from a given TAR file.

### CODE REVIEW

**Comment by lvalerom** [comment]
Good morning,

We encountered a minor problem with the specification of this new feature. We have already prepared a fix for it. We were wondering if it would be better to push the changes to this pull request or wait until this one is closed and open a new one later.

Please advise.

Regards

**Comment by keflavich** [comment]
Add it to this PR.

File: astroquery/esa/xmm_newton/core.py
Comment (keflavich) [inline astroquery/esa/xmm_newton/core.py:318]:
```suggestion
        Examples
        --------
```

File: astroquery/esa/xmm_newton/core.py
Comment (keflavich) [inline astroquery/esa/xmm_newton/core.py:320]:
```suggestion
        Extract all bands and instruments::
```

File: astroquery/esa/xmm_newton/core.py
Comment (keflavich) [inline astroquery/esa/xmm_newton/core.py:324]:
```suggestion
        If we want to retrieve the band 3 for the instrument PN::
```

Comment (keflavich) [reply astroquery/esa/xmm_newton/core.py:324]: What does "PN" mean?  Define acronyms in the docs

File: astroquery/esa/xmm_newton/core.py
Comment (keflavich) [inline astroquery/esa/xmm_newton/core.py:329]:
```suggestion
        Extract the exposure and detector maps::
```

File: astroquery/esa/xmm_newton/core.py
Comment (keflavich) [inline astroquery/esa/xmm_newton/core.py:327]:
```suggestion
        ``fits_image`` will be the full path to the extracted FTZ file
```

File: astroquery/esa/xmm_newton/core.py
Comment (keflavich) [inline astroquery/esa/xmm_newton/core.py:335]:
```suggestion
        If we want to retrieve exposure map in the band 3 for the instrument PN::
```

File: astroquery/esa/xmm_newton/core.py
Comment (keflavich) [inline astroquery/esa/xmm_newton/core.py:338]:
```suggestion
        For retrieving the detector mask in the band 3 for the instrument PN::
```

File: astroquery/esa/xmm_newton/core.py
Comment (keflavich) [inline astroquery/esa/xmm_newton/core.py:421]: This should just raise an exception.

**Comment by keflavich** [comment]
Also, acronyms need to be defined.  In particular, EPIC is not defined anywhere.

This feature should be included in the narrative docs too (`docs/esa/xmm_newton.rst`)

**Comment by lvalerom** [comment]
Good morning,

Thank your for your feedback, I believe I've fixed everything that was suggested. 

Please let me know if there are any additional changes you advise.

Thank you
Luis

File: astroquery/esa/xmm_newton/core.py
Comment (keflavich) [inline astroquery/esa/xmm_newton/core.py:435]: When I said to raise an error, I meant just skip the whole try/except; it's not needed.
```suggestion
        with tarfile.open(filename, "r") as tar:
            for i in tar.getmembers():
                paths = os.path.split(i.name)
                fname = paths[1]
                paths = os.path.split(paths[0])
                if paths[1] != "pps":
                    continue
                fname_info = self._parse_filename(fname)
                if fname_info["X"] != "P":
                    continue
                if not fname_info["I"] in instrument:
                    continue
                if not int(fname_info["S"]) in band:
                    continue
                if not fname_info["T"] in _product_type:
                    continue
                tar.extract(i, _path)
                if not ret.get(int(fname_info["S"])):
                    ret[int(fname_info["S"])] = {}
                b = int(fname_info["S"])
                ins = fname_info["I"]
                value = os.path.abspath(os.path.join(_path, i.name))
                if fname_info["T"] == "DETMSK":
                    ins = fname_info["I"] + "_det"
                elif fname_info["T"] == "EXPMAP":
                    ins = fname_info["I"] + "_expo"
                if ret[b].get(ins) and type(ret[b].get(ins)) == str:
                    log.warning("More than one file found with the "
                                "band %u and "
                                "the instrument: %s" % (b, ins))
                    ret[b][ins] = [ret[b][ins], value]
                elif ret[b].get(ins) and type(ret[b].get(ins)) == list:
                    ret[b][ins].append(value)
                else:
                    ret[b][ins] = value
```

Comment (lvalerom) [reply astroquery/esa/xmm_newton/core.py:435]: Good morning,

Quick note. Regarding the moving of this feature to an utils.py, we have prepared two similar features that extract files from a given tar file and it would make sense to also put them there. However, one of them reads content of specific files and does queries to an FTP server to download extra data. So, would you still advise this feature to be put in this module?

Thank you
Luis

**Comment by keflavich** [comment]
No, if an additional query needs to be done, keep that in the main module, probably.  Could you describe a little more what the intended use case is for that?  Why would you need an additional file when un-taring a local file?

**Comment by lvalerom** [comment]
In that particular case, the PPS tar file does not include the Response Matrix File (RMF) needed for the analysis of the spectral products. I'm not sure as to the reason why they did not include this file there but the only way to retrieve the appropriate RMF is to download it from an FTP server.

**Comment by lvalerom** [comment]
Good morning @keflavich 

Are there any updates regarding the suggestion of moving this feature to `utils.py`? Please let me know if you need anything else from my side.

Thank you

File: astroquery/esa/xmm_newton/core.py
Comment (keflavich) [inline astroquery/esa/xmm_newton/core.py:315]:
```suggestion
        Additionally, ``get_detmask`` and ``get_exposure_map`` can be set to True.
```

**Comment by bsipocz** [comment]
@keflavich - I do a final check and rebase, there are plenty of unrelated commits present.

**Comment by bsipocz** [comment]
I've cleaned up the commit history and rebased, once travis passed this can be merged, after adding a changelog.

**Review by bsipocz — APPROVED** [review]
I've added the changelog to get it merged

**Comment by bsipocz** [comment]
Thanks @lvalerom!

**Pull request merged**
